From cf71644ac87d4723fa1af63a7d1fabbb7bad6b28 Mon Sep 17 00:00:00 2001
Message-ID: <cf71644ac87d4723fa1af63a7d1fabbb7bad6b28.1698444193.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1698444193.git.sharlatanus@gmail.com>
References: <cover.1698444193.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 23 Oct 2023 01:44:21 +0100
Subject: [PATCH 04/19] gnu: python-spherical-geometry: Enable tests.

* gnu/packages/astronomy.scm (python-spherical-geometry): Enable tests.
[build-system]: Switch to pyproject-build-system.
[arguments]{phases}: Add 'build-extensions phase.
---
 gnu/packages/astronomy.scm | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 18cf562e12..3b85d4ec9e 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -2470,19 +2470,22 @@ (define-public python-spherical-geometry
        (file-name (git-file-name name version))
        (sha256
         (base32 "0kzcncqir4v7nhk9lxj9gxr32p3krkaqa58y2i4kksgxxy24qw4z"))))
-    (build-system python-build-system)
+    (build-system pyproject-build-system)
     (arguments
      (list
-      ;; NOTE: (Sharlatan-20220523T231348+0100): Tests depends on old Python2
-      ;; libarry `sphere'
-      #:tests? #f
+      ;; XXX: Disable one failing test
+      ;; See https://github.com/spacetelescope/spherical_geometry/issues/252
+      #:test-flags #~(list "-k" "not test_overlap")
       #:phases
       #~(modify-phases %standard-phases
           (add-after 'unpack 'preparations
             (lambda _
               (setenv "SETUPTOOLS_SCM_PRETEND_VERSION" #$version)
               ;; Use our own libraries in place of bundles.
-              (setenv "USE_SYSTEM_QD" "1"))))))
+              (setenv "USE_SYSTEM_QD" "1")))
+          (add-before 'check 'build-extensions
+            (lambda _
+              (invoke "python" "setup.py" "build_ext" "--inplace"))))))
     (native-inputs
      (list python-pytest
            python-setuptools-scm))
-- 
2.41.0

